src/main/java/com/passbook/sparkeighteen/service/TransactionService.java

```diff
@@ -128,4 +127,43 @@ public String deleteTransaction(Integer transactionID) {
         }
         return "Entered Transaction ID does not Exists";
     }
-}
\ No newline at end of file
+
+    /**
+     * This updateTransaction method is used to update the existing transaction in the repository.
+     * @param transactionID is to specify which transaction the user wants to update.
+     * @param request is for transaction details to update their transaction.
+     * @return the transaction response whether transaction is updated or is empty.
+     */
+    public TransactionResponse updateTransaction(Integer transactionID, @Valid TransactionRequest request) {
+
+        if (request.getAmount() <= 0)
+            return TransactionResponse.builder()
+                    .message("Invalid Transaction amount")
+                    .build();
+
+        Optional<TransactionEntity> userTransaction = transactionRepository.findById(transactionID);
+        if (userTransaction.isEmpty()) {
+            return TransactionResponse.builder()
+                    .message("Transaction does not exist.")
+                    .txnID(transactionID)
+                    .build();
+        }
+        final TransactionEntity transaction = userTransaction.get();
+
+        transaction.setNote(request.getNote());
+
+        final TransactionEntity saveTransaction = transactionRepository.save(transaction);
+
+        return TransactionResponse.builder()
+                .txnID(transactionID)
```

Comment:
be consistent and use saveTransaction 
I know it will be same only but be consistent.